lib/segment/src/index/field_index/geo_hash.rs

```diff
@@ -198,6 +199,28 @@ fn check_intersection(geohash: &str, circle: &GeoRadius) -> bool {
         > bbox_center.haversine_distance(&Point::new(circle.center.lon, circle.center.lat))
 }
 
+/// Check if geohash tile intersects the polygon
+fn check_polygon_intersection(geohash: &str, polygon: &GeoPolygon) -> bool {
+    let precision = geohash.len();
+    if precision == 0 {
+        return true;
+    }
+    let rect = decode_bbox(geohash).unwrap();
+
+    #[allow(deprecated)]
```

Comment:
Could we simply use the non deprecated `Coord` API here?

```rust
use geo::Coord;

    let polygon_ring: Vec<Coord<f64>> = polygon
        .points
        .iter()
        .map(|point| Coord {
            x: point.lon,
            y: point.lat,
        })
        .collect();
```

Reply:
Yes I was wondering why we use Coordinate in our code base? I see some places deliberately added. If that is okay, I can replace all deprecated Coordinate with Coord

https://github.com/qdrant/qdrant/blob/904e0ef5a1ff0eb94ac07b4b9ae38acbd0b65dfb/lib/segment/src/index/field_index/geo_hash.rs#L4C1-L4C2